### PR TITLE
Add test executor

### PR DESCRIPTION
**Pull request by mschuwalow**
part of #169 

As we implement more parts of the API we can enable the tests one by one.

Also fixes a bug in the PING command

### CODE REVIEW

File: redis/src/main/scala/zio/redis/Interpreter.scala [hunk 1]
```diff
@@ -75,6 +77,27 @@ trait Interpreter {
           .tapError(e => logger.error(s"Executor exiting: $e"))
     }
 
+    private final class InMemory(
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/Interpreter.scala:80]: I'd remove `()`.

File: redis/src/main/scala/zio/redis/Interpreter.scala [hunk 2]
```diff
@@ -75,6 +77,27 @@ trait Interpreter {
...
+    ) extends Service {
+      override def execute(command: Chunk[RespValue.BulkString]): zio.IO[RedisError, RespValue] =
+        for {
+          name   <- ZIO.fromOption(command.headOption).mapError(_ => ProtocolError("Malformed command."))
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/Interpreter.scala:84]:
```suggestion
          name   <- ZIO.fromOption(command.headOption).orElseFail(ProtocolError("Malformed command."))
```

File: redis/src/main/scala/zio/redis/Interpreter.scala [hunk 3]
```diff
@@ -86,6 +109,9 @@ trait Interpreter {
           } yield live
       }
 
+    val inMemory: ZLayer[Any, Nothing, RedisExecutor] =
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/Interpreter.scala:112]: I'd rename this to `test`.

File: redis/src/main/scala/zio/redis/Interpreter.scala (hunk 1 above)
Comment (mschuwalow) [reply redis/src/main/scala/zio/redis/Interpreter.scala:80]: Makes sense for now. I was thinking that it will have STM data structures in it

Comment (mijicd) [reply redis/src/main/scala/zio/redis/Interpreter.scala:80]: Ah yes, true. Well, you'll introduce it anyhow, so this is a no-op.

**Review by mijicd — APPROVED** [review]

**Pull request merged**